### PR TITLE
Added Datasets, Transforms, and one base model type

### PR DESCRIPTION
**Pull request by ab7289**

### CODE REVIEW

File: src/datasets.py [hunk 1]
```diff
@@ -66,3 +67,50 @@ def get_dataset(
         return train_dataset, valid_dataset, test_dataset
     else:
         raise ValueError("Invalid Dataset name")
+
```
Comment (os0000os) [inline src/datasets.py:70]: `train_len = int(len(train_dataset) * valid_ratio)` should be `train_len = int(len(train_dataset) * (1-valid_ratio))` I think. (Sorry I couldn't figure out how to add this comment to the correct line)

File: src/datasets.py [hunk 2]
```diff
@@ -66,3 +67,50 @@ def get_dataset(
...
+        else:
+            raise ValueError("Invalid number of indices for batch_size")
+    else:
+        train_batch, valid_batch, test_batch = batch_size, batch_size, batch_size
+    return (
+        DataLoader(
+            train_dataset, batch_size=train_batch, shuffle=True, drop_last=drop_last
+        ),
+        DataLoader(
+            valid_dataset, batch_size=valid_batch, shuffle=True, drop_last=drop_last
```
Comment (os0000os) [inline src/datasets.py:111]: I think it's more common **not** to shuffle the VALID and TEST data? Am guessing doing so reduces the noise when comparing model to model.

File: src/datasets.py [hunk 3]
```diff
@@ -66,3 +67,50 @@ def get_dataset(
...
+        ),
+        DataLoader(
+            test_dataset, batch_size=test_batch, shuffle=True, drop_last=drop_last
```
Comment (os0000os) [inline src/datasets.py:114]: (Same as above)

File: src/datasets.py [hunk 4]
```diff
@@ -1,10 +1,11 @@
...
+from typing import Tuple, Union
 
 import torch_geometric.datasets as datasets
 import torch_geometric.transforms as T
 from torch.utils.data import random_split
 from torch_geometric.data import Dataset
+from torch_geometric.loader import DataLoader
 from torch_geometric.transforms import base_transform
 
 _DATASET_NAMES = dict(
```
Comment (os0000os) [inline src/datasets.py:11]: Minor suggestion: perhaps rename `get_dataset` as `get_datasets` to (1) be consistent with `get_dataloaders` and (2) suggest to caller that it'll return more than one dataset objects? (Again, sorry I couldn't figure out how to comment to the correct line.)

File: src/datasets.py (hunk 1 above)
Comment (ab7289) [reply src/datasets.py:70]: oops you're right!

File: src/datasets.py (hunk 2 above)
Comment (ab7289) [reply src/datasets.py]: Oh good idea!

File: src/models.py [hunk 5]
```diff
@@ -0,0 +1,135 @@
...
+            x, batch = gnn.max_pool_x(cluster, x, batch)
+
+        if self.readout == "mean":
+            x = gnn.global_mean_pool(x, batch)
+        elif self.readout == "max":
+            x = gnn.global_max_pool(x, batch)
+        elif self.readout == "add":
+            x = gnn.global_add_pool(x, batch)
+
+        F.dropout(x, p=self.dropout, training=self.training)
```
Comment (os0000os) [inline src/models.py:131]: Should the output be assigned to x? `x = F.dropout(x, ...)`

File: src/transforms.py [hunk 6]
```diff
@@ -4,30 +4,50 @@
 def get_transforms_list(type: str = "cartesian"):
     if type is "default":
```
Comment (os0000os) [inline src/transforms.py:5]: A minor question: since the default parameter value `"cartesian"` is not implemented otherwise, would it be better to change this `if` statement to `if type is "cartesian"`?

File: src/models.py (hunk 5 above)
Comment (ab7289) [reply src/models.py]: Good eye! Fixed

File: src/transforms.py (hunk 6 above)
Comment (ab7289) [reply src/transforms.py]: good catch!

File: src/utils.py [hunk 7]
```diff
@@ -34,6 +34,10 @@ def normalized_cut_2d(edge_index: torch.Tensor, pos: torch.Tensor):
...
-def calculate_accuracy(outputs: torch.Tensor, targets: torch.Tensor):
-    pred = outputs.max(1)[1]
-    return pred.eq(targets).sum().item()
+def calculate_accuracy(outputs: torch.Tensor, targets: torch.Tensor) -> float:
+    """
+    calculates the accuracy between the argmax of the last dimension of outputs
+    and the targets
+    """
+    pred = outputs.argmax(-1)
+    return pred.view_as(targets).eq(targets).sum().item()
```
Comment (os0000os) [inline src/utils.py:43]: I think I found the bug: the returned should have been divided by the size of the batch::  return `pred.view_as(targets).eq(targets).sum().item() / targets.numel()`

Comment (ab7289) [reply src/utils.py:43]: ooh ty! i'll try it out!

File: tests/conftest.py [hunk 8]
```diff
@@ -0,0 +1,33 @@
...
+    )
+    return train_dataset, valid_dataset, test_dataset
+
+
+@pytest.fixture
+def test_graph_input():
+    data = Data(
+        x=torch.ones((10, 1)),
+        edge_index=torch.ones((2, 15), dtype=torch.long),
+        edge_attt=torch.ones((15, 2)),
```
Comment (os0000os) [inline tests/conftest.py:29]: Not sure if it matters - the parameter should be `edge_attr`.

**Review by os0000os — APPROVED** [review]
Other than a minor typo in `test_graph_input()`, all look good to me.
